### PR TITLE
[PBM-1405] dump v2

### PR DESCRIPTION
**Pull request by defbin**

### CODE REVIEW

File: pbm/archive/conv.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+	"strings"
+
+	mtArchive "github.com/mongodb/mongo-tools/common/archive"
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/bson"
+
+	"github.com/percona/percona-backup-mongodb/pbm/storage"
+)
+
+func GenerateV1FromV2(stg storage.Storage, bcp, rs string) error {
```
Comment (boris-ilijic) [inline pbm/archive/conv.go:18]: Do we have the Jira ticket to align backup and restore about the metadata format? So currently we use V2->V1 conversion. In future we should make backup to export directly to V1, or we should make restore to work with V2 format directly.

File: pbm/archive/conv.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	if err != nil {
+		return errors.Wrapf(err, "open file: %q", metaV2Filename)
+	}
+	defer rdr.Close()
+
+	data, err := io.ReadAll(rdr)
+	if err != nil {
+		return errors.Wrap(err, "read")
+	}
+	rdr.Close()
```
Comment (boris-ilijic) [inline pbm/archive/conv.go:30]:
```suggestion
	err = rdr.Close()
	if err != nil {
		return errors.Wrap(err, "close")
	}
```
It'll be cleaner to add error handling here.

**Review by boris-ilijic — COMMENTED** [review]
This looks good to me, please see few comments.

File: pbm/archive/backup.go [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+
+func (bcp *backupImpl) writeMeta(collections []*NamespaceV2) error {
+	meta := &ArchiveMetaV2{
+		Version:       version.Current().Version,
+		ServerVersion: bcp.serverVersion,
+		FCV:           bcp.fcv,
+		Namespaces:    collections,
+	}
+
+	data, err := bson.Marshal(meta)
```
[the remark this answers is not in the document]
Comment (defbin) [reply pbm/archive/backup.go:336]: BSON is a better format for the data (direct support for binary (ObjectID), int64 (CRC, size), and compact storage size). JSON is text-only format. the meta file is for PBM only. not for end user. IMO, for testing, using bsondump is not a big deal (it is needed rarely)

File: pbm/archive/conv.go (hunk 1 above)
Comment (defbin) [reply pbm/archive/conv.go]: v1 or v2 - does not matter. converting to v1 was done as quick solution. it is pbm format. when PBM creates archive for mongorestore, it uses data from the v1 but not format directly. PBM can makes the archive from v2. just some changes are needed in another place.

**Pull request merged**